flang/lib/Lower/Bridge.cpp

```diff
@@ -2659,8 +2665,32 @@ class FirConverter : public Fortran::lower::AbstractConverter {
     }
     analyzeExplicitSpacePop();
   }
+
   void analyzeExplicitSpacePop() { explicitIterSpace.popLevel(); }
 
+  void addMaskVariable(Fortran::lower::FrontEndExpr exp) {
+    // Note: use i8 to store bool values. This avoids round-down behavior found
+    // with sequences of i1. That is, an array of i1 will be truncated in size
+    // and be too small. For example, a buffer of type fir.array<7xi1> will have
+    // 0 size.
```

Comment:
Is this a FIR/MLIR bug or just that i1 array are a not something that should be used in general ?

Reply:
I would say it is a bug. I am not sure where the fault lies.